metrics/ci_integrations/lib/firestore/adapter/firestore_storage_client_adapter.dart

```diff
@@ -26,12 +26,9 @@ class FirestoreStorageClientAdapter implements StorageClient {
       final futures = <Future>[];
 
       for (final build in builds) {
-        futures.add(collection.add(
-          build.toJson()
-            ..addAll(
-              {'projectId': project.id},
-            ),
-        ));
+        final map = build.toJson();
+        map['projectId'] = project.id;
```

Comment:
I think we should build.copyWith(projectId: project.id) now as we don't want to keep toJson details (like field name) here.